### PR TITLE
Decoupled QnAMakerDialog from preview libs

### PR DESCRIPTION
**Pull request by lauren-mills**
Changes:
- Moved QnAMakerActionBuilder and QnADialogResponseOptions into AI.QnA and changed dependencies on ActivityTemplates to using Activities
- Added QnAMakerDialog to AI.QnA.Dialogs that doesn't depend on Expressions or ActivityTemplates

### CODE REVIEW

**Comment by fuselabs** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Builder/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.Luis.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.Luis/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.QnA.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.QnA/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.ApplicationInsights.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.ApplicationInsights/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Azure.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Azure/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Dialogs.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Dialogs/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.TemplateManager.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.TemplateManager/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Configuration.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Configuration/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Connector.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Connector/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Schema.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Schema/4.5.3)

File: libraries/Microsoft.Bot.Builder.AI.QnA/QnAMakerActionBuilder.cs [hunk 1]
```diff
@@ -15,26 +15,26 @@ namespace Microsoft.Bot.Builder.AI.QnA
     /// <summary>
     /// QnAMaker action builder class.
     /// </summary>
-    internal class QnAMakerActionBuilder
+    public class QnAMakerActionBuilder
```
Comment (tomlm) [inline libraries/Microsoft.Bot.Builder.AI.QnA/QnAMakerActionBuilder.cs]: >public [](start = 4, length = 6)

I think we don't want to expose this class, so keep this as 'internal' and add internal visible to attribute for AdaptiveDialog like this:
#if SIGNASSEMBLY
[assembly: InternalsVisibleTo("Microsoft.Bot.Builder.Dialogs.Adaptive, PublicKey=0024000004800000940000000602000000240000525341310004000001000100b5fc90e7027f67871e773a8fde8938c81dd402ba65b9201d60593e96c492651e889cc13f1415ebb53fac1131ae0bd333c5ee6021672d9718ea31a8aebd0da0072f25d87dba6fc90ffd598ed4da35e44c398c454307e8e33b8426143daec9f596836f97c8f74750e5975c64e2189f45def46b2a2b1247adc3652bf5c308055da9")]
#else
[assembly: InternalsVisibleTo("Microsoft.Bot.Builder.Dialogs.Adaptive")]
#endif

This should give permission to Adaptive Dialog assembly to use the internal QnAMakerActionBuilder class

File: libraries/Microsoft.Bot.Builder.AI.QnA/Dialogs/QnAMakerDialog.cs [hunk 2]
```diff
@@ -0,0 +1,160 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
```
Comment (cleemullins) [inline libraries/Microsoft.Bot.Builder.AI.QnA/Dialogs/QnAMakerDialog.cs]: There is no file called "QnAMakerDialogTests.cs". This file needs a corresponding test file. 

I do see a "Microsoft.Bot.Builder.AI.QnA.Tests/QnAMakerTests.cs" file, but that is full of Adaptive tests even it's testing this class.

File: tests/Microsoft.Bot.Builder.AI.QnA.Tests/QnAMakerTests.cs [hunk 3]
```diff
@@ -14,6 +14,7 @@
 using Microsoft.Bot.Builder.Dialogs.Adaptive;
```
Comment (cleemullins) [inline tests/Microsoft.Bot.Builder.AI.QnA.Tests/QnAMakerTests.cs]: These are all Adaptive tests. They is not an adaptive assembly. The QnaMakerDialog class isn't (that I can tell) in the Adaptive namespace. 

This layering probably isn't quite right.

**Comment by cleemullins** [comment]
Core comments:
1. The tests also need to be decoupled from adaptive. 
2. The ctor for the QnA Maker Dialog seems really complex. 
3. A JS implementation is needed before we can merge.

**Pull request closed without merging**